src/components/structures/MatrixChat.js

```diff
@@ -126,6 +143,33 @@ module.exports = React.createClass({
         }
     },
 
+    _registerAsGuest: function() {
+        var self = this;
+        var config = this.props.config;
+        console.log("Doing guest login on %s", config.default_hs_url);
+        MatrixClientPeg.replaceUsingUrls(
+            config.default_hs_url, config.default_is_url
+        );
+        MatrixClientPeg.get().registerGuest().done(function(creds) {
```

Comment:
This seems to just call registerGuest() & see if it works, but my understanding was that registerGuest may require user interactive auth just like a regular register (but probably with a looser set of auth requirements).